release notes/v0.40.0.md

```diff
@@ -0,0 +1,285 @@
+      }
+      else if (msg.event === 'ERROR') {
+        console.error(`VU ${__VU}:${id} received:: ${msg.message}`)
+      }
+      else {
+        console.log(`VU ${__VU}:${id} received unhandled message: ${msg.message}`)
+      }
+    })
+
+
+    let intervalId = setInterval(() => {
+      ws.send(JSON.stringify({ 'event': 'SAY', 'message': `I'm saying ${randomString(5)}` }));
+    }, randomIntBetween(2000, 8000)); // say something every 2-8seconds
+
+
+    let timeout1id = setTimeout(function() {
+      clearInterval(intervalId)
+      console.log(`VU ${__VU}:${id}: ${sessionDuration}ms passed, leaving the chat`);
+      ws.send(JSON.stringify({ 'event': 'LEAVE' }));
+    }, sessionDuration);
+
+    let timeout2id = setTimeout(function() {
+      console.log(`Closing the socket forcefully 3s after graceful LEAVE`);
+      ws.close();
+    }, sessionDuration + 3000);
+
+    ws.addEventListener("close", () => {
+      clearTimeout(timeout1id);
+      clearTimeout(timeout2id);
+      console.log(`VU ${__VU}:${id}: disconnected`);
+    })
+  });
+}
+```
+</details>
+
+Note that no k6 iterations finish if any WebSocket is still open or if a timeout or an interval is not cleared or triggered. This means that your script must take care of clearing all intervals and closing the WebSocket at some point. However, k6 still kills the whole process if it takes too long to stop after the maximum test duration is reached.
+
+Current issues and future improvements for the WebSockets API can be found in its [issue tracker](https://github.com/grafana/xk6-websockets/issues). Currently, documentation is available through [MDN](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket), though some features are not yet supported:
+- no Blob binary type - ArrayBuffer is the default
+- no `onMessage` and co. - only addEventListener
+
+### First-class support for JavaScript Classes
+
+As part of [updating goja](https://github.com/grafana/k6/pull/2610), k6 got native support for classes. Again, that's native, as in not by transpiling by the internal Babel.
+
+Because this actually implements classes as described in the latest ECMAScript specification, this also means we get a ton of additional class features that were never previously supported (for example, private fields). Additionally, at least one bug [#1763](https://github.com/grafana/k6/issues/1763) was fixed as a result of this, but probably many more as well.
+
+Due to this fairly significant change, some code could behave differently. Please report any issues, though consider that it's possible that the new behavior is just the correct one.
+
+Other updates from goja are:
+- optimizations around using strings and some access patterns
+- support for `\u{01234}` Unicode point syntax in regexp
+- Fixed a case where interrupting the VM did not work, especially around try/catch usage. This was particularly problematic for k6, as it could lead to k6 hanging. [#2600](https://github.com/grafana/k6/pull/2600)
```

Comment:
```suggestion
- Fixed a case where interrupting the VM did not work, especially around try/catch usage ([#2600](https://github.com/grafana/k6/pull/2600)). This was particularly problematic for k6, as it could lead to k6 hanging.
```